src/test/java/com/heatheryou/httpserver/ResponseBuilderTest.java

```diff
@@ -1,60 +1,58 @@
-import com.heatheryou.httpserver.constants.EntityHeader;
 import org.junit.Before;
 import org.junit.Test;
 
 import static com.heatheryou.httpserver.constants.StatusLine.STATUS_CODE_200;
-import static org.junit.Assert.assertEquals;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.*;
 
 public class ResponseBuilderTest {
+    String[] entityHeaders;
     ResponseBuilder builder;
+    Response response;
 
     @Before
     public void setUp() {
+        entityHeaders = new String[]{"Content-Length: 0"};
         builder = new ResponseBuilder();
+        response = builder.buildResponse(STATUS_CODE_200, entityHeaders, CharacterSet.EMPTY);
     }
 
     @Test
-    public void setStatusLineSetsTheStatusLineGivenStatusCode() {
-        builder.setStatusLine(STATUS_CODE_200);
+    public void buildResponseReturnsAResponseObject() {
+        assertThat(response, instanceOf(Response.class));
+    }
+
+    @Test
+    public void getStatusLineReturnsStatusLine() {
         String actual = builder.getStatusLine();
-        String expected = "HTTP/1.1 200 OK" + CharacterSet.CRLF;
+        String expected = "HTTP/1.1 200 OK\r\n";
         assertEquals(expected, actual);
     }
 
     @Test
-    public void setEntityHeaderSetsTheEntityHeadersWhenEntityHeadersArePassedInAsStringArray() {
-        String contentLength = EntityHeader.CONTENT_LENGTH + CharacterSet.SPACE + EntityHeader.CONTENT_LENGTH_0 + CharacterSet.CRLF;
-        String[] entityArray = new String[]{contentLength};
-        builder.setEntityHeader(entityArray);
+    public void getEntityHeaderReturnsEntityHeader() {
         String actual = builder.getEntityHeader();
-        String expected = "Content-Length: 0" + CharacterSet.CRLF;
+        String expected = "Content-Length: 0";
         assertEquals(expected, actual);
     }
 
     @Test
-    public void getHeaderReturnsFullHeaderWhenStatusLineAndEntityHeadersAreSet() {
-        builder.setStatusLine(STATUS_CODE_200);
-        String contentLength = EntityHeader.CONTENT_LENGTH + CharacterSet.SPACE + EntityHeader.CONTENT_LENGTH_0 + CharacterSet.CRLF;
-        String[] entityArray = new String[]{contentLength};
-        builder.setEntityHeader(entityArray);
-        builder.setHeader();
+    public void getHeaderReturnsStatusLineAndEntityHeader() {
         String actual = builder.getHeader();
         String expected = String.join("\r\n", new String[]{
                 "HTTP/1.1 200 OK",
-                "Content-Length: 0",
-                ""
+                "Content-Length: 0"
         });
         assertEquals(expected, actual);
     }
 
     @Test
-    public void getBodyReturnsEmptyBodyWhenThereIsNoBody() {
-        builder.setBody(CharacterSet.EMPTY);
+    public void getBodyReturnsBody() {
         String actual = builder.getBody();
-        String expected = CharacterSet.CRLF;
+        String expected = "\r\n";
         assertEquals(expected, actual);
```

Comment:
This feels like an odd test since your example data doesn't actually have a body. :wink:

Reply:
😝 You're right. I removed it for now and will test it again when I have the POST request working. e272622d6dc851292693d0fbfce322494b713eba